test/modules/layers/test_quantisation.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+
+import torch
+from torch import nn
+from torchmultimodal.modules.layers.quantisation import Quantisation
+from torchmultimodal.utils.preprocess import (
+    flatten_to_channel_vectors,
+    reshape_from_channel_vectors,
+)
+
+
+class TestQuantisation(unittest.TestCase):
+    """
+    Test the Quantisation class
+    """
+
+    def setUp(self):
+        torch.set_printoptions(precision=10)
+        torch.manual_seed(4)
+        self.num_embeddings = 4
+        self.embedding_dim = 5
+        self.encoded = torch.randn((2, self.embedding_dim, 3, 3))
+        self.embedding_weights = torch.randn((self.num_embeddings, self.embedding_dim))
+
+    def test_quantised_output(self):
+        vq = Quantisation(
+            num_embeddings=self.num_embeddings, embedding_dim=self.embedding_dim
+        )
+        vq.embedding = nn.Embedding.from_pretrained(self.embedding_weights)
+        x, permuted_shape = flatten_to_channel_vectors(self.encoded, 1)
+        output = vq(x)
```

Comment:
why do we need these processing? cant we hard code the tensor format to begin with?

Reply:
The quantization layer assumes the input tensor is already "flattened" into a shape where the last dimension is the number of embeddings (which is all what matters here, more details see comments in the module). This allows generalization to all modalities.

Reply:
can we assume that all modalities' input data will have the dimensions batch x channel x n dims? I understood that to be consistent across pytorch

Reply:
How about text data where "channel" is embedding dim? Is it always the second in the shape array? 
The question about input tensor shape is orthogonal to our delegating the flattening / unflattening to separate functions anyways.

Reply:
That's fine as long as we keep throwing the exception inside `forward` that verifies last dim of the permuted shape is equal to embedding dim of the quantized vector. My concern is that we don't always have control over what happens inside the encoder despite the convention.